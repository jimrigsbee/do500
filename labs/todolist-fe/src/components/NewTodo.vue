<template>
    <md-field>
      <label>New ToDo</label>
      <md-input :placeholder="placeholderMsg"
        @keyup.enter="newTodoAdded"
        v-model="newTodo"
      ></md-input>
    </md-field>
  </template>
<script>
export default {
  name: "NewTodo",
  props: {
    placeholderMsg: String
  },
  data() {
    return {
      newTodo: ""
    };
  },
  methods: {
    newTodoAdded(e) {
      console.info("INFO - Adding new todo ", this.newTodo);
      this.newTodo = e.target.value;
      this.$store.dispatch("setNewTodo", this.newTodo);
      this.$store.dispatch("addTodo");
      this.$store.dispatch("clearNewTodo");
      this.newTodo = "";
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
</style>
