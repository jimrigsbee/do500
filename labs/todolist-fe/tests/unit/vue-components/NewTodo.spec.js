/* eslint-disable */
import { shallow, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import NewTodo from "@/components/NewTodo.vue";

import * as all from "../setup.js";

const localVue = createLocalVue();

localVue.use(Vuex);
let methods;
let store;

describe("NewTodo.vue", () => {
  beforeEach(() => {
    methods = {
      newTodoAdded: jest.fn()
    };
    store = new Vuex.Store({
      state: {},
      methods
    });
  });

  it("calls newTodoAdded() when keyup.enter hit.", () => {
    // time to try and test some vuex stuff and see if the methods are called when expected.
    const wrapper = shallow(NewTodo, { methods, localVue });
    const input = wrapper.find(".md-input");
    input.trigger("keyup.enter");
    expect(methods.newTodoAdded).toHaveBeenCalled();
  });

  it("does not call newTodoAdded() when keyup.space hit.", () => {
    // time to try and test some vuex stuff and see if the methods are called when expected.
    const wrapper = shallow(NewTodo, { methods, localVue });
    const input = wrapper.find(".md-input");
    input.trigger("keyup.space");
    expect(methods.newTodoAdded).not.toHaveBeenCalled();
  });

  it("renders props.placeholderMsg when passed", () => {
    const msg = "Add a Todo";
    const wrapper = shallow(NewTodo, {
      propsData: { placeholderMsg: msg }
    });
    expect(wrapper.vm._props.placeholderMsg).toMatch(msg);
  });

  it("renders newTodo as empty string", () => {
    const wrapper = shallow(NewTodo, {});
    expect(wrapper.vm.newTodo).toMatch("");
  });

  it("has the expected html structure", () => {
    const wrapper = shallow(NewTodo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
