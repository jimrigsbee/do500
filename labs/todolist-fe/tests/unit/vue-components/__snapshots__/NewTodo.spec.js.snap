// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodo.vue has the expected html structure 1`] = `<!---->`;
